src/FsAutoComplete.Core/FakeSupport.fs

```diff
@@ -2,8 +2,96 @@ namespace FsAutoComplete
 
 open Fake.Runtime
 open Fake.Runtime.Trace
+open System
+open System.IO
+open System.IO.Compression
+open System.Threading.Tasks
+open System.Net
+open Newtonsoft.Json.Linq
 
 module FakeSupport =
+  let private compatibleFakeVersion = "5.15.0"
+
+  let private fakeDownloadUri version =
+    sprintf "https://github.com/fsharp/FAKE/releases/download/%s/fake-dotnetcore-portable.zip" version
+    |> Uri
+
+  let downloadAndExtract (uri:Uri) directory = async {
+    use client = new WebClient()
+    client.Headers.Add("user-agent", "Ionide")
+    let tempFile = Path.GetTempFileName()
+    try
+      do! client.DownloadFileTaskAsync(uri, tempFile) |> Async.AwaitTask
+      ZipFile.ExtractToDirectory(tempFile, directory)
+    finally
+      try
+        File.Delete tempFile
+      with e -> ()
+  }
+
+  let private downloadAndGetFakeDll uri directory = async {
+    let fakeDll = Path.Combine(directory, "fake.dll")
+    if File.Exists fakeDll then
+      return fakeDll
+    else    
+      do! downloadAndExtract uri directory
+      if not (File.Exists fakeDll) then
+        failwithf "No 'fake.dll' in '%s' after downloading fake" directory
+      return fakeDll
+  }
+
+  let mutable private fakeDll = None
+  let private locker = obj()
+  let private getFakeRuntimeAsTask() =
+    let installRuntime () =
+      // use ~/.fsac if possible
+      let usersDir = Environment.GetFolderPath Environment.SpecialFolder.UserProfile
+      let fsac = System.IO.Path.Combine(usersDir, ".fsac")
```

Comment:
or `.ionide`?

Reply:
I think we may try using `.ionide` for a while - it's something that people already have in `.gitignore`.

Probably in the future we should replace all `.ionide` with `.fsac` but for now let's be consistent.

Reply:
Note this is the users folder, not the project, so gitignore doesn't matter

Reply:
Oh, right. Yes, `.fsac` is good choice in such case.